src/metabase/notification/seed.clj

```diff
@@ -1,124 +1,178 @@
-                            :recipients      [{:type    :notification-recipient/template
-                                               :details {:pattern     "{{context.admin_email}}"
-                                                         :is_optional true}}
-                                              {:type                 :notification-recipient/group
-                                               :permissions_group_id (:id (perms-group/admin))}]}]}]))
-
-(def ^:private notification-related-models
-  [:model/Notification
-   :model/NotificationHandler
-   :model/NotificationSubscription
-   :model/NotificationRecipient
-   :model/ChannelTemplate])
-
-(defn- truncate-table!
-  "Truncate a table and restart the identity column."
-  [table-name]
-  (case (db/db-type)
-    :postgres
-    (t2/query {:truncate [table-name :restart :identity :cascade]})
-    :mysql
-    (do
-      (t2/delete! table-name)
-      (t2/query (format "ALTER TABLE `%s` AUTO_INCREMENT = 1" (name table-name))))
-    (do
-      (t2/delete! table-name)
-      (t2/query {:alter-table [table-name {:alter-column [:id :restart :with 1]}]}))))
-
-(defn- truncate-notification-related-tables!
-  []
-  (doseq [model notification-related-models
-          :let [table (t2/table-name model)]]
-    (log/infof "Truncating table %s" table)
-    (truncate-table! table)))
+          {:internal_id   "system-event/slack-token-error"
+           :active        true
+           :payload_type  :notification/system-event
+           :subscriptions [{:type       :notification-subscription/system-event
+                            :event_name :event/slack-token-invalid}]
+           :handlers      [{:active       true
+                            :channel_type :channel/email
+                            :channel_id   nil
+                            :template     {:name         "Slack Token Error Email template"
+                                           :channel_type "channel/email"
+                                           :details      {:type "email/handlebars-resource"
+                                                          :subject "Your Slack connection stopped working"
+                                                          :path "metabase/email/slack_token_error.hbs"
+                                                          :recipient-type "cc"}}
+                            :recipients   [{:type    :notification-recipient/template
+                                            :details {:pattern "{{context.admin_email}}" :is_optional true}}
+                                           {:type                 :notification-recipient/group
+                                            :permissions_group_id (:id (perms-group/admin))}]}]}]))
+
+
+(defn- cleanup-notification!
+  [internal-id existing-row]
+  (t2/delete! :model/Notification :internal_id internal-id)
+  (when-let [template-ids (->> existing-row :handlers (map (comp :id :template)) (filter some?) not-empty)]
```

Comment:
Can use `keep` instead of `map` then `filter`

Reply:
nice!